ROOTFS/usr/bin/waggle-firewall

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash -e
+
+# Print the script usage help
+print_help() {
+  echo """
+usage: $0 [-r]
+
+Adds or removes firewall rules.
+
+  -r : deletes firewall rules instead of adding them
+  -? : print this help menu
+"""
+}
+
+DEL=
+# Process script input arguments
+while getopts "r?" opt; do
+  case $opt in
+    r) DEL=1
+      ;;
+    ?|*)
+      print_help
+      exit 1
+      ;;
+  esac
+done
+
+CONFIGS=("wlan-interface" "wifi-interface" "modem-interface")
+RULES=(
+	"INPUT -i {{INTF}} -m conntrack ! --ctstate RELATED,ESTABLISHED -j DROP"
+	"FORWARD -i {{INTF}} -m conntrack ! --ctstate RELATED,ESTABLISHED -j DROP"
+)
+
+INTF=()
+for config in "${CONFIGS[@]}"; do
+	VAL=$(waggle-get-config -s hardware -k $config)
+	if [ $? -eq 0 ]; then
+		INTF+=("$VAL")
+	fi
+done
+
+for r in "${RULES[@]}"; do
+	for i in "${INTF[@]}"; do
+		rule=$(echo $r | sed "s/{{INTF}}/$i/g")
+		# test if the rule is already applied
+		if [ $(iptables-save | grep -c "$rule") -eq 0 ]; then
```

Comment:
This looks good to me.

In case you have a preference, I think iptables also has a -C check flag letting you write write something like:

```sh
if ! iptables -C $rule; then
  iptables -I $rule
fi
```

Reply:
sounds good! done.